crates/bevy_math/src/cubic_splines.rs

```diff
@@ -306,50 +321,64 @@ impl<P: Point> CubicGenerator<P> for CubicBSpline<P> {
 /// Error during construction of [`CubicNurbs`]
 #[derive(Debug, Error)]
 pub enum CubicNurbsError {
-    /// Provided knot vector had an invalid length.
-    #[error("Invalid knot vector length: expected {expected}, provided {provided}")]
-    InvalidKnotVectorLength {
-        /// Expected knot vector length
+    /// Provided the wrong number of knots.
+    #[error("Wrong number of knots: expected {expected}, provided {provided}")]
+    KnotsMismatch {
+        /// Expected number of knots
```

Comment:
```suggestion
    #[error("Wrong number of knots: expected {expected}, provided {provided}")]
    KnotsNumberMismatch {
        /// Expected number of knots
```
I think this variant name should be more clear that the thing that mismatches is the number of knots.

Reply:
This change would also require to change the usage